src/etc/unicode.py

```diff
@@ -307,12 +307,114 @@ def emit_table(f, name, t_data, t_type = "&'static [(char, char)]", is_pub=True,
     format_table_content(f, data, 8)
     f.write("\n    ];\n\n")
 
+def emit_trie_lookup_range_table(f):
+    f.write("""
+pub struct BoolTrie {
```

Comment:
It'd be great to have an high-level overview of what all these fields represent/how it actually represents the data.